### ISSUE TITLE
react-native init x --template file:///x fail to add dependencies

### ISSUE THREAD

**Issue by codler** [facebook/react-native#20198]
## Environment
Run `react-native info` in your terminal and paste its contents here.

  React Native Environment Info:
    System:
      OS: macOS High Sierra 10.13.5
      CPU: x64 Intel(R) Core(TM) i7-4770HQ CPU @ 2.20GHz
      Memory: 545.44 MB / 16.00 GB
      Shell: 2.7.1 - /usr/local/bin/fish
    Binaries:
      Node: 8.9.4 - /usr/local/bin/node
      Yarn: 1.6.0 - /usr/local/bin/yarn
      npm: 5.6.0 - /usr/local/bin/npm
      Watchman: 4.9.0 - /usr/local/bin/watchman
    SDKs:
      iOS SDK:
        Platforms: iOS 11.4, macOS 10.13, tvOS 11.4, watchOS 4.3
    IDEs:
      Android Studio: 3.1 AI-173.4720617
      Xcode: 9.4.1/9F2000 - /usr/bin/xcodebuild
    npmPackages:
      react: 16.4.1 => 16.4.1
      react-native: 0.56.0 => 0.56.0
    npmGlobalPackages:
      react-native-cli: 2.0.1
      react-native-git-upgrade: 0.2.7

## Description
I have created a local react-native-template and have a dependencies.json file, the dependencies never gets when initializing new project

    react-native init x --template file:///x

## Reproducible Demo
Create a folder with correct values in package.json and dependencies.json,
Run `react-native init x --template file:///path-to-folder`

```
Adding dependencies for the project...
No additional dependencies.
Adding develop dependencies for the project...
No additional develop dependencies.
error Not enough arguments, expected at least 1.
Failed to clean up template temp files in node_modules/. This is not a critical error, you can work on your app.
```

**Comment by hramos** [comment]
The `--template` flag expects a suffix which will be used to look up a `react-native-template-{suffix}` package in npm.

**Comment by codler** [comment]
Here it says you can use file:///
https://medium.com/dailyjs/the-1-2-3s-of-react-native-templates-1f5dda037e11

**Comment by hramos** [comment]
Good to know! I can see in `local-cli/generator/templates.js` that this is indeed the case.

We did land a change recently that might help (https://github.com/facebook/react-native/commit/5219585ef90f7a9ade66fdc0d9362a5706dfcc8b) but it's not in a release just yet. I haven't yet attempted to reproduce this issue, but anyone interested in doing so should start by looking at the file I referenced in the first sentence above.
